### PR TITLE
[Fabric-Admin] Set remote bridge after reverse pair the bridge device

### PR DESCRIPTION
**Pull request by yufengwangca**
After complete fabric sync setup process from E1 to E2, we not only need to setup the remote fabric bridge in E1, but also need to setup the remote bridge in E2. Otherwise, we can only sync device from E2 to E1, not the other direction.

I don't think we have test case cover this scenario right now, that is why it is not found. https://github.com/project-chip/connectedhomeip/issues/36450 to track this

### CODE REVIEW

File: examples/fabric-admin/device_manager/DeviceSynchronization.cpp [hunk 1]
```diff
@@ -142,13 +142,19 @@ void DeviceSynchronizer::OnReportEnd()
 
 void DeviceSynchronizer::OnDone(app::ReadClient * apReadClient)
 {
+    ChipLogProgress(NotSpecified, "Synchronization complete for NodeId:" ChipLogFormatX64, ChipLogValueX64(mNodeId));
+
 #if defined(PW_RPC_ENABLED)
-    if (mState == State::ReceivedResponse && !DeviceManager::Instance().IsCurrentBridgeDevice(mNodeId))
+    // We need to register the remote bridge device as a bridged device on the local bridge. This ensures that the ecosystem
```
Comment (tehampson) [inline examples/fabric-admin/device_manager/DeviceSynchronization.cpp:148]: Not sure I am following what you mean with `We need to register the remote bridge device as a bridged device on the local bridge.`

Why are we registering the remote bridge on the local bridge. I interpret this to mean you are listing the remote bridge as an endpoint on the local bridge. Where in the spec does it say we have to do this?

File: examples/fabric-admin/device_manager/DeviceSynchronization.cpp [hunk 2]
```diff
@@ -142,13 +142,19 @@ void DeviceSynchronizer::OnReportEnd()
...
+    // triggering the fabric sync process is notified of the successful reverse commissioning.
+    if (mState == State::ReceivedResponse && (mIsReverseCommissioning || !DeviceManager::Instance().IsCurrentBridgeDevice(mNodeId)))
     {
+        mIsReverseCommissioning = false;
```
Comment (tehampson) [inline examples/fabric-admin/device_manager/DeviceSynchronization.cpp:152]: Why add this flag if we can ask `DeviceManager` if the node id is the remote bridge? (Note if that functionality is not there we can add it in).

I also don't understand why we want to enumerate the remote bridge as an endpoint on the local bridge. What part of the spec are you trying to implement here. If you are not implementing something from the spec, what is missing from the spec that you are trying to address here?

File: examples/fabric-admin/device_manager/DeviceSynchronization.cpp (hunk 1 above)
Comment (yufengwangca) [reply examples/fabric-admin/device_manager/DeviceSynchronization.cpp:148]: There are two reasons.

1. If the side which trigger the FS setup process want to get confirmed if the receiving side has successfully reverse paired its bridge, we need this to notify the state change by registering the bridge as the bridged device. But this is not mandate by the spec, it is optional.

2. The primary reason is I found you used this mechanism to test non-bridge device pairing in our test harness, such as:
https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TC_MCORE_FS_1_3.py#L221
The th_server_th_node_id is paired via the reverse commissing of CCTRL to automate the test. Some of the test won't pass without this change.

File: examples/fabric-admin/device_manager/DeviceSynchronization.cpp (hunk 2 above)
Comment (yufengwangca) [reply examples/fabric-admin/device_manager/DeviceSynchronization.cpp:152]: See, the reason above, the current TH needs this and current implementation did that,  this change is to keep backward compatibility

File: examples/fabric-admin/device_manager/DeviceSynchronization.cpp (hunk 1 above)
Comment (tehampson) [reply examples/fabric-admin/device_manager/DeviceSynchronization.cpp:148]: >But this is not mandate by the spec, it is optional.

Then why are we writing an example that relies on this. I don't think this is the right direction to go in. While it solves your problem today this is not solved at scale when ecosystems try to actually sync together.

>https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TC_MCORE_FS_1_3.py#L221

This is **NOT** checking that a remote bridge is added. This is a check that TH_SERVER is added not a remote bridge. This smells like something isn't right to me and you are relying on something else to make these test pass in a way we are not.

Comment (yufengwangca) [reply examples/fabric-admin/device_manager/DeviceSynchronization.cpp:148]: Yes, it checks if the TH_SERVER is added by comparing the endpoint, but it used the reverse commission to pair the TH_SERVER. I don't think it is appropriate, but I don't know why we implemented the cert test like this, if we want to undo this, we need to discuss how to undo it on 1.4 branch

File: examples/fabric-admin/device_manager/DeviceSynchronization.cpp (hunk 2 above)
Comment (yufengwangca) [reply examples/fabric-admin/device_manager/DeviceSynchronization.cpp:152]: Per offline talking, we plan to update the current cert tests to not using the reverse commission to commission the test server, we need to hole this PR until we update all relevant cert tests

File: examples/fabric-admin/device_manager/DeviceSynchronization.cpp (hunk 1 above)
Comment (yufengwangca) [reply examples/fabric-admin/device_manager/DeviceSynchronization.cpp:148]: Per offline talking, we plan to update the current cert tests to not using the reverse commission to commission the test server, we need to hole this PR until we update all relevant cert tests

Comment (yufengwangca) [reply examples/fabric-admin/device_manager/DeviceSynchronization.cpp]: Removed this logic to add device as a bridged device after reverse pairing

File: examples/fabric-admin/rpc/RpcServer.cpp [hunk 3]
```diff
@@ -147,11 +174,12 @@ class FabricAdmin final : public rpc::FabricAdmin, public IcdManager::Delegate
         if (error == CHIP_NO_ERROR)
         {
             NodeId nodeId = DeviceManager::Instance().GetNextAvailableNodeId();
```
Comment (tehampson) [inline examples/fabric-admin/rpc/RpcServer.cpp:176]: nit, why not just assign to mNodeId directly. Is it possible that before `DeviceManager::Instance().PairRemoteDevice(nodeId, code.c_str());` is called that mNodeId changes? If not I would remove nodeId

**Review by tehampson — APPROVED** [review]

**Review by joonhaengHeo — APPROVED** [review]

**Pull request merged**